File: rle_decompress.cs
// rle_decompress.cs
using System;
using System.IO;
using System.Text;
using System.Collections.Generic;

class RLEDecompress
{
    static string Colorize(string text, string color)
    {
        string col = color switch
        {
            "green" => "\x1b[92m",
            "red" => "\x1b[91m",
            "yellow" => "\x1b[93m",
            _ => "\x1b[0m"
        };
        return col + text + "\x1b[0m";
    }

    const char ESCAPE = '\\';

    static string Decompress(string text)
    {
        if (string.IsNullOrEmpty(text)) return "";
        var result = new StringBuilder();
        int i = 0, n = text.Length;
        while (i < n)
        {
            char ch = text[i];
            if (ch == ESCAPE)
            {
                if (i + 1 < n && text[i + 1] == ESCAPE)
                {
                    result.Append(ESCAPE);
                    i += 2;
                    continue;
                }
                if (i + 1 >= n) throw new Exception("Unexpected end after escape");
                char repeatChar = text[i + 1];
                i += 2;
                string numStr = "";
                while (i < n && char.IsDigit(text[i]))
                {
                    numStr += text[i];
                    i++;
                }
                if (string.IsNullOrEmpty(numStr)) throw new Exception("Missing number after escape");
                int count = int.Parse(numStr);
                result.Append(repeatChar, count);
            }
            else
            {
                result.Append(ch);
                i++;
            }
        }
        return result.ToString();
    }

    static string ReadInput(string filename)
    {
        if (string.IsNullOrEmpty(filename) || filename == "-")
        {
            return Console.In.ReadToEnd();
        }
        return File.ReadAllText(filename, Encoding.UTF8);
    }

    static void WriteOutput(string filename, string content)
    {
        if (string.IsNullOrEmpty(filename) || filename == "-")
        {
            Console.Write(content);
        }
        else
        {
            File.WriteAllText(filename, content, Encoding.UTF8);
        }
    }

    static void Main(string[] args)
    {
        string inputFile = null, outputFile = null;
        bool verbose = false;
        for (int i = 0; i < args.Length; i++)
        {
            if (args[i] == "-v" || args[i] == "--verbose")
            {
                verbose = true;
            }
            else if (inputFile == null)
            {
                inputFile = args[i];
            }
            else
            {
                outputFile = args[i];
            }
        }
        if (inputFile == null)
        {
            Console.WriteLine(Colorize("Usage: rle_decompress <input> [output] [-v]", "yellow"));
            return;
        }

        string data;
        try
        {
            data = ReadInput(inputFile);
        }
        catch (Exception e)
        {
            Console.WriteLine(Colorize("Error reading input: " + e.Message, "red"));
            return;
        }
        int inputSize = Encoding.UTF8.GetByteCount(data);

        string result;
        try
        {
            result = Decompress(data);
        }
        catch (Exception e)
        {
            Console.WriteLine(Colorize("Decompression error: " + e.Message, "red"));
            return;
        }
        int outputSize = Encoding.UTF8.GetByteCount(result);

        if (verbose)
        {
            double ratio = inputSize > 0 ? (double)outputSize / inputSize : 1.0;
            Console.WriteLine(Colorize($"Compressed size: {inputSize} bytes", "yellow"));
            Console.WriteLine(Colorize($"Decompressed size: {outputSize} bytes", "yellow"));
            Console.WriteLine(Colorize($"Expansion ratio: {ratio:F2}x", "green"));
        }

        try
        {
            WriteOutput(outputFile, result);
            if (!string.IsNullOrEmpty(outputFile) && outputFile != "-")
            {
                Console.WriteLine(Colorize("Result written to " + outputFile, "green"));
            }
        }
        catch (Exception e)
        {
            Console.WriteLine(Colorize("Error writing output: " + e.Message, "red"));
        }
    }
}

File: rle_decompress.java
// rle_decompress.java
import java.io.*;
import java.nio.charset.StandardCharsets;
import java.nio.file.*;
import java.util.*;

public class rle_decompress {
    private static final String RESET = "\u001B[0m";
    private static final String GREEN = "\u001B[92m";
    private static final String RED = "\u001B[91m";
    private static final String YELLOW = "\u001B[93m";

    private static String colorize(String text, String color) {
        return color + text + RESET;
    }

    private static final char ESCAPE = '\\';

    private static String decompress(String text) throws Exception {
        if (text == null || text.isEmpty()) return "";
        StringBuilder result = new StringBuilder();
        int i = 0, n = text.length();
        while (i < n) {
            char ch = text.charAt(i);
            if (ch == ESCAPE) {
                if (i + 1 < n && text.charAt(i + 1) == ESCAPE) {
                    result.append(ESCAPE);
                    i += 2;
                    continue;
                }
                if (i + 1 >= n) throw new Exception("Unexpected end after escape");
                char repeatChar = text.charAt(i + 1);
                i += 2;
                StringBuilder numStr = new StringBuilder();
                while (i < n && Character.isDigit(text.charAt(i))) {
                    numStr.append(text.charAt(i));
                    i++;
                }
                if (numStr.length() == 0) throw new Exception("Missing number after escape");
                int count = Integer.parseInt(numStr.toString());
                for (int k = 0; k < count; k++) result.append(repeatChar);
            } else {
                result.append(ch);
                i++;
            }
        }
        return result.toString();
    }

    private static String readInput(String filename) throws IOException {
        if (filename == null || filename.equals("-")) {
            StringBuilder sb = new StringBuilder();
            try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
                String line;
                while ((line = reader.readLine()) != null) {
                    sb.append(line).append('\n');
                }
            }
            return sb.toString();
        }
        return new String(Files.readAllBytes(Paths.get(filename)), StandardCharsets.UTF_8);
    }

    private static void writeOutput(String filename, String content) throws IOException {
        if (filename == null || filename.equals("-")) {
            System.out.print(content);
        } else {
            Files.write(Paths.get(filename), content.getBytes(StandardCharsets.UTF_8));
        }
    }

    public static void main(String[] args) {
        List<String> argList = Arrays.asList(args);
        String inputFile = null, outputFile = null;
        boolean verbose = false;
        for (String arg : argList) {
            if (arg.equals("-v") || arg.equals("--verbose")) {
                verbose = true;
            } else if (inputFile == null) {
                inputFile = arg;
            } else {
                outputFile = arg;
            }
        }
        if (inputFile == null) {
            System.out.println(colorize("Usage: java rle_decompress <input> [output] [-v]", YELLOW));
            return;
        }

        String data;
        try {
            data = readInput(inputFile);
        } catch (Exception e) {
            System.err.println(colorize("Error reading input: " + e.getMessage(), RED));
            return;
        }
        int inputSize = data.getBytes(StandardCharsets.UTF_8).length;

        String result;
        try {
            result = decompress(data);
        } catch (Exception e) {
            System.err.println(colorize("Decompression error: " + e.getMessage(), RED));
            return;
        }
        int outputSize = result.getBytes(StandardCharsets.UTF_8).length;

        if (verbose) {
            double ratio = inputSize > 0 ? (double) outputSize / inputSize : 1.0;
            System.out.println(colorize("Compressed size: " + inputSize + " bytes", YELLOW));
            System.out.println(colorize("Decompressed size: " + outputSize + " bytes", YELLOW));
            System.out.println(colorize(String.format("Expansion ratio: %.2fx", ratio), GREEN));
        }

        try {
            writeOutput(outputFile, result);
            if (outputFile != null && !outputFile.equals("-")) {
                System.out.println(colorize("Result written to " + outputFile, GREEN));
            }
        } catch (Exception e) {
            System.err.println(colorize("Error writing output: " + e.getMessage(), RED));
        }
    }
}
